salt/modules/kubernetes.py

```diff
@@ -622,6 +622,58 @@ def show_namespace(name, **kwargs):
         _cleanup()
 
 
+def set_container_image(image, deployment,
+                         container,
+                         namespace="default",
+                         **kwargs):
+    '''
+    Set the image of a container in the kubernetes deployment defined
+    by the specified name.
+
+    CLI Examples::
+
+        salt '*' kubernetes.set_deployment_image nginx:latest nginx-deployment nginx-container namespace=prod
+
+    '''
+    _setup_conn(**kwargs)
+
+    try:
+        api_instance = kubernetes.client.ExtensionsV1beta1Api()
+        patch = {
+            "spec":{
+                "template":{
+                    "spec":{
+                        "containers":[
+                            {
+                                "name":container,
+                                "image":image
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+        api_response = api_instance.patch_namespaced_deployment(
+            name=deployment,
+            namespace=namespace,
+            body=patch)
+
+        return api_response.to_dict()
+
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception(
+                'Exception when calling '
+                'ExtensionsV1beta1Api->patch_namespaced_deployment: '
+                '{0}'.format(exc)
+            )
+            raise CommandExecutionError(exc)
```

Comment:
1. There is no need to `return None`, in Python especially.
2. Your `log.exception` isn't done right: please use `%s` syntax.
3. Single exit is more welcome than multiple returns.
4. You use hard-coded 404 instead of constants from library.

So I would write it as follows:
```python
def set_container_image(.....
    ret = None
    try:
        ...
        ret = api_response.to_dict()
    except (ApiException, HTTPError) as err:
        if isinstance(err, HTTPError) and err.status != http.client.NOT_FOUND:
            log.exception('HTTP error: %s', six.text_type(err))
            raise CommandExecutionError(err)
    return ret
```